add a sourcemap option to frida-compile to enable/disable sourcemap generation

cosmosgenius opened the pull request:
This PR adds a debug option to `frida-compile` which gets passed on to browserify.
The debug option disables/enables sourcemap on the generated file. 

The reason to have such an option is to reduce the generated file's size. You could easily reduce approximately 33% file size by disabling sourcemap.

oleavr commented on bin/compile.js:
```diff
@@ -15,6 +15,7 @@ program
     .option('-w, --watch', 'watch for changes and recompile')
     .option('-b, --bytecode', 'output bytecode')
     .option('-x, --no-babelify', 'skip Babel transforms')
+    .option('-d, --debug', 'enable debug in browserify')
```
This is a breaking change, as the current default is to include it. How about `-S, --no-sourcemap` or something like that?

oleavr commented on index.js:
```diff
@@ -317,6 +316,9 @@ function writeFile(file, data, options) {
 
 function trimSourceMap(molder) {
   var map = molder.sourcemap;
+  if(!map) {
```
For consistency with the existing coding style this should look like this:

```js
  if (map === undefined)
    return '';
```

(or `null`, don't know which it is in this case.)

cosmosgenius commented:
I have updated the commit to have `-S` instead of `-d` defaulting to generation of sourcemap. Updated the if condition to match the coding style. 
The `map` values is indeed '`undefined` when sourcemap is not there. Let me know if anythng else is required.

oleavr commented on bin/compile.js:
```diff
@@ -15,6 +15,7 @@ program
     .option('-w, --watch', 'watch for changes and recompile')
     .option('-b, --bytecode', 'output bytecode')
     .option('-x, --no-babelify', 'skip Babel transforms')
+    .option('-S, --no-sourcemap', 'disable sourcemap in browserify')
```
How about `'omit sourcemap'`? I think it's better to avoid mentioning internal implementation details.

cosmosgenius replied:
updated the comment

oleavr commented on bin/compile.js:
```diff
@@ -29,6 +30,7 @@ const options = {
   target: program.target,
   bytecode: !!program.bytecode,
   babelify: program.babelify,
+  sourcemap: !!program.sourcemap,
```
Is this correct after the switch got renamed?

cosmosgenius replied:
yes adding --no prefix inverts it via commander.

oleavr replied:
Ahh right! I guess we don't need the `!!` to ensure a boolean here btw? (Seeing as we don't do the conversion for `babelify`.)

cosmosgenius replied:
Updated(removed !!)
